runtime/meta/ffi.cpp

```diff
@@ -166,6 +206,11 @@ extern "C" {
     set_len(rvalue, rtype->size);
     free(avalues);
 
+    for (auto &s : structTypes) {
+      free(s->elements);
+      free(s);
+    }
+
```

Comment:
don't forget to call structTypes.clear() here otherwise you'll have a double free when you call this method a second time.